src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,94 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private Object[] array;
+    private int capacity = 10;
+    private int size = 0;
+
+    public ArrayList() {
+        array = new Object[capacity];
+    }
 
     @Override
     public void add(T value) {
-
+        if (size == capacity) {
+            grow();
+        }
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            grow();
+        }
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
+    }
 
+    private void grow() {
+        capacity = capacity + capacity / 2;
+        Object[] tmp = new Object[capacity];
+        System.arraycopy(array, 0, tmp, 0, size);
+        array = tmp;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index > size - 1) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        return (T) array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index > size - 1) {
```

Comment:
What if less than zero?
```suggestion
        if (index >= size) {
```